cmd/chunks-inspect/loki.go

```diff
@@ -122,6 +126,31 @@ func parseLokiChunk(chunkHeader *ChunkHeader, r io.Reader) (*LokiChunk, error) {
 		return nil, fmt.Errorf("failed to read rawData for Loki chunk into memory: %w", err)
 	}
 
+	c, _ := chunkenc.NewByteChunk(data, 0, 0)
+	bs := c.Blocks(from, through)
+	fmt.Println("Blocks: ", len(bs))
+
+	pipeline := log.NewNoopPipeline()
+	for idx, block := range bs {
+		fmt.Println("Block  : ", idx)
+		fmt.Println("MinTime: ", time.Unix(0, block.MinTime()).UTC())
+		fmt.Println("MaxTime: ", time.Unix(0, block.MaxTime()).UTC())
+		fmt.Println("Offset : ", block.Offset())
+		iter := block.Iterator(context.Background(), pipeline.ForStream(nil))
+		for iter.Next() {
+			e := iter.Entry()
+			fmt.Println(e.Timestamp.UTC(), " ", e.Line)
+			for _, meta := range e.StructuredMetadata {
+				fmt.Println("\t", meta.Name, "=", meta.Value)
+			}
+		}
+	}
+
+	err := c.Close()
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
this seems like a reasonable change, but looking at the structure of the rest of the code I think printing of all these details would be done in `main.go`, it looks to me like process just builds the details of the chunk into something that `main.go` prints

I haven't been keeping up with structured metadata. If it's separate from a chunk then imo we should either have a separate function to gather and return the structured metadata to `main.go` or an additional return value in `parseLokiChunk` for the metadata, perhaps with a parameter to choose whether or not you want to receive info about the structured metadata.

Reply:
Thank you @cstyan for your feedback! I will restructure the code and go forward.